src/grammar.py

```diff
@@ -40,25 +42,50 @@ def interpret(self, gdfs):
 
 
 @dataclass(frozen=True, repr=False)
-class Dissolve(GrammarRule):
-    gdf: GDF
+class Dissolve(Candidate):
+    gdf: str
+    _: KW_ONLY
     by: str
 
     def __repr__(self):
         return f'{self.gdf}.dissolve(by="{self.by}")'
 
     def interpret(self, gdfs):
-        gdf = self.gdf.interpret(gdfs)
-        return gdf.dissolve(by=self.by)
+        return gdfs[self.gdf].dissolve(by=self.by)
 
 
 @dataclass(frozen=True, repr=False)
-class SJoin(GrammarRule):
+class SJoin(Candidate):
     left: str
     right: str
+    _: KW_ONLY
+    how: str
+    predicate: str
```

Comment:
To help us remember what the literal values we support in our synthesizer, could we specify these types explicitly as their string literal unions, e.g.:

```suggestion
    how: Literal["left", "right", "inner"]
    predicate: Literal["intersects", "within", "contains", "overlaps", "crosses", "touches"]
```

This will be especially helpful down the road to keep track of what parts of the geopandas API we do an don't support in the synthesizer.

Reply:
Great idea!